pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -56,73 +56,75 @@ func (c *Reconciler) createOrUpdateAffinityAssistants(ctx context.Context, wb []
 	var errs []error
 	var unschedulableNodes sets.Set[string] = nil
 	for _, w := range wb {
-		if w.PersistentVolumeClaim != nil || w.VolumeClaimTemplate != nil {
-			affinityAssistantName := getAffinityAssistantName(w.Name, pr.Name)
-			a, err := c.KubeClientSet.AppsV1().StatefulSets(namespace).Get(ctx, affinityAssistantName, metav1.GetOptions{})
-			claimName := getClaimName(w, *kmeta.NewControllerRef(pr))
-			switch {
-			// check whether the affinity assistant (StatefulSet) exists or not, create one if it does not exist
-			case apierrors.IsNotFound(err):
-				affinityAssistantStatefulSet := affinityAssistantStatefulSet(affinityAssistantName, pr, claimName, c.Images.NopImage, cfg.Defaults.DefaultAAPodTemplate)
-				_, err := c.KubeClientSet.AppsV1().StatefulSets(namespace).Create(ctx, affinityAssistantStatefulSet, metav1.CreateOptions{})
+		if w.PersistentVolumeClaim == nil && w.VolumeClaimTemplate == nil {
+			continue
+		}
+
+		var claimTemplates []corev1.PersistentVolumeClaim
+		var claims []corev1.PersistentVolumeClaimVolumeSource
+		if w.PersistentVolumeClaim != nil {
+			claims = append(claims, *w.PersistentVolumeClaim.DeepCopy())
+		} else if w.VolumeClaimTemplate != nil {
+			claimTemplate := w.VolumeClaimTemplate.DeepCopy()
+			// PVCs Will be created by Affinity Assistant StatefulSet and will follow the naming format `<claimTemplate.Name>-<AffinityAssistantName>-0`
+			claimTemplate.Name = volumeclaim.GetPVCNameWithoutAffinityAssistant(w.VolumeClaimTemplate.Name, w, *kmeta.NewControllerRef(pr))
+			claimTemplates = append(claimTemplates, *claimTemplate)
+		}
```

Comment:
Just curious, this code is inside of the for loop, why do we need an array to append claims and claimTemplates?

Reply:
Oh I see, thanks!